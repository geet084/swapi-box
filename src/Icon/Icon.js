import React from 'react'
import './Icon.scss'

const Icon = ({url, choice, classes, getInfo}) => {
  
  return (
    <div className="section">
      <input type="image"
        src={url}
        alt={choice}
        onClick={getInfo}
        className={classes}
        name={choice}
      />
    <span className="tag">{choice}</span>
    </div>
  )
}

export default Icon;