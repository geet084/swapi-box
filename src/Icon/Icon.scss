input:focus {
    outline:none;
}

.section {
  display: flex;
  flex-direction: column;
}

.tag {
  margin: 20px 0 -20px 8px;
  color: white;
}

.active {
  cursor: pointer;
  background-color: yellow;
  border-radius: 100px;
  padding: 5px;
  transition-duration: .3s;
  margin: -5px;
  &:hover {
    cursor: pointer;
    background-color: green;
    border-radius: 100px;
    padding: 5px;
    transition-duration: 1.5s;
    margin: -5px;
  }
}

.inactive {
  transition-duration: .1s;
  &:hover {
    cursor: pointer;
    background-color: white;
    border-radius: 100px;
    padding: 5px;
    transition-duration: .1s;
    margin: -5px;
  }
}

