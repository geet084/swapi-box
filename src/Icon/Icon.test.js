import React from 'react'
import { shallow } from 'enzyme'
import Icon from './Icon'

describe('Icon', () => {
  let wrapper
  let mockURL = ''
  let mockIcon = ''
  let mockGetInfo = jest.fn()
  let mockClasses = ''

  beforeEach(() => {
    wrapper = shallow(
      <Icon
        url={mockURL}
        choice={mockIcon}
        getInfo={mockGetInfo}
        classes={mockClasses}
      />
    )
  })

  it('should properly render default snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should register a click', () => {
    wrapper.find('input').simulate('click')
    expect(mockGetInfo).toHaveBeenCalled()
  })
})