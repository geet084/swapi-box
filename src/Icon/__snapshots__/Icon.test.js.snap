// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should properly render default snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Icon
    choice=""
    classes=""
    getInfo={[MockFunction]}
    url=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          alt=""
          className=""
          name=""
          onClick={[MockFunction]}
          src=""
          type="image"
        />,
        <span
          className="tag"
        >
          
        </span>,
      ],
      "className": "section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": "",
          "name": "",
          "onClick": [MockFunction],
          "src": "",
          "type": "image",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            "",
          ],
          "className": "tag",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          "",
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            alt=""
            className=""
            name=""
            onClick={[MockFunction]}
            src=""
            type="image"
          />,
          <span
            className="tag"
          >
            
          </span>,
        ],
        "className": "section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "",
            "name": "",
            "onClick": [MockFunction],
            "src": "",
            "type": "image",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              "",
            ],
            "className": "tag",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            "",
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
